Tensile/Source/lib/source/ContractionProblem.cpp

```diff
@@ -1049,6 +1050,42 @@ namespace Tensile
                 TENSILE_ASSERT_EXC(DataTypeInfo::Get(m_d.dataType()).isComplex);
     }
 
+    void ContractionProblem::calcArithmeticIntensity()
+    {
+        size_t problemSize = 1;
+        for(size_t i = 0; i < m_problemSizes.size(); ++i)
+        {
+            problemSize *= m_problemSizes[i];
+        }
+        double gflop = 2 * problemSize * 1e-9;
+
+        size_t aSize = 1;
+        for(size_t i = 0; i < m_a.dimensions(); ++i)
+        {
+            aSize *= m_a.sizes()[i];
+        }
+        size_t bSize = 1;
+        for(size_t i = 0; i < m_b.dimensions(); ++i)
+        {
+            bSize *= m_b.sizes()[i];
+        }
+        size_t cSize = 1;
+        for(size_t i = 0; i < m_c.dimensions(); ++i)
+        {
+            cSize *= m_c.sizes()[i];
+        }
+        if(m_beta != 0) // If problem includes beta, update gflops and gbytes
+        {
+            gflop += 2 * cSize * 1e-9; // Include (+ beta * C) in gflops
+            cSize *= 2; // Include read C and write D in gbytes
```

Comment:
This looks a little strange.
D is irrelevant to beta.

Reply:
This accounts for read C and write D if beta is non-zero. Otherwise it just needs gbytes for write D